CodeStarterCamp_Week5/main.swift

```diff
@@ -8,5 +8,139 @@
 
 import Foundation
 
-print("Hello, World!")
+enum Level {
+    case A, B, C
+}
 
+protocol Talent {
+    var singing: Level? { get }
+    var dancing: Level? { get }
+    var acting: Level? { get }
+    
+    func haveAGrade() -> Bool
+}
+
+protocol BadPersonality{
+    var frequancyOfCursing: Level? { get }
+    
+    func haveBadPersonality() -> Bool
+}
+
+class Person {
+    let name: String
+    let height: Int
+    
+    init(name: String, height: Int) {
+        self.name = name
+        self.height = height
+    }
+}
+
+class TalentedPerson: Person, Talent {
+    var singing: Level?
+    var dancing: Level?
+    var acting: Level?
+    
+    init(name: String, height: Int, singing: Level, dancing: Level, acting: Level) {
+        super.init(name: name, height: height)
+        
+        self.singing = singing
+        self.dancing = dancing
+        self.acting = acting
+    }
+    
+    func haveAGrade() -> Bool {
+        guard let siningLv = self.singing, let dancingLv = self.dancing, let actingLv = self.acting else { return false }
```

Comment:
```suggestion
        guard let siningLv = self.singing, 
              let dancingLv = self.dancing, 
              let actingLv = self.acting else { return false }
```
`guard`로 여러 구문을 한 번에 처리할 때는 이런 컨벤션을 사용하기도 합니다!

Reply:
옵셔널을 없애면서 필요 없는구문이라 우선 삭제했습니다 추후 참고하겠습니다